slapo/pattern.py

```diff
@@ -9,5 +9,14 @@ def forward(self, *args):
         raise NotImplementedError
 
 
+class CallModule(nn.Module):
```

Comment:
The names of CallModule and call_module are confusing.

Reply:
They are just used in different places. `CallModule` is used in a pattern class, while `call_module` can be used in a pattern function. Do you have any suggestions about the naming?

Reply:
Hmm base on your use cases, such as

```python
    class LinearReLUPattern(slapo.Pattern):
        def __init__(self):
            super().__init__()
            self.fc = CallModule(r"fc?")
            self.relu = nn.ReLU()
```

Here you're actually constructing a module instead of calling it, so I guess you could use the name like `ModulePattern` or something like that.

Reply:
It may also lead to confusion with the `Pattern` class.

Reply:
Well, to me it is a kind of pattern, especially you have a regex in its argument.

Reply:
Yes, I agree it is a pattern, and `ModulePattern` is a proper name, but since we have the `Pattern` base class which is also a module, I'm afraid it will cause confusion between these two.